hudi-common/src/main/java/org/apache/hudi/common/table/log/HoodieMergedLogRecordScanner.java

```diff
@@ -110,6 +117,17 @@ protected <T> void processNextRecord(HoodieRecord<T> newRecord) throws IOExcepti
     }
   }
 
+  /**
+   * Processing time based delete is found when
+   * 1. The current record is a delete whose orderingVal is default value, or
+   * 2. The current record's metadata contains the flag: PROCESSING_TIME_BASED_DELETE_FOUND.
+   */
+  private <T> boolean hasProcessingTimeBasedDelete(HoodieRecord<T> record) throws IOException {
+    return (record.isDelete(readerSchema, getPayloadProps())
```

Comment:
can we flip the comparison. lets first check (record.getMetadata().isPresent() && record.getMetaDataInfo(PROCESSING_TIME_BASED_DELETE_FOUND).isPresent()
and then check for ordering value from the record.

Reply:
yes

Reply:
Done